smart-orchestrator infrastructure failure: default-workflow step-07-write-tests Parse error: unexpected token: LBracket

## Repro

Invoke smart-orchestrator on a docs-only task (e.g., resolving issue #896 — README rewrite). After step-06c-documentation-refinement, default-workflow advances to step-07-write-tests and fails immediately with:

\`\`\`
[ERROR] Step 'execute-single-round-1' failed: Sub-recipe 'default-workflow' failed.
Failed steps:
- step-07-write-tests [error]: Condition error: Parse error: unexpected token: LBracket
\`\`\`

Completed 17 of 18 steps before this failure (workflow preparation through documentation refinement).

## Impact

- Docs-only changes (no test step needed) cannot ship through default-workflow.
- The conditional that gates step-07-write-tests appears to have a parser bug rejecting an LBracket token.
- Required adaptive strategy: bypass smart-orchestrator and make the docs change directly (announced per skill instructions).

## Asks

1. Fix the condition parser in default-workflow's step-07-write-tests gate so docs-only tasks can either skip the step gracefully or run with no-op tests.
2. Add a regression check exercising default-workflow on a docs-only task description.

## Environment

- amplihack home: ~/.amplihack
- recipe runner invoked from /home/azureuser/src/Simard/worktrees/main
- AMPLIHACK_AGENT_BINARY=copilot
- failed run timestamp: 2026-04-18T11:14:07Z

--- a/src/amplihack/recipes/rust_runner_binary.py
+++ b/src/amplihack/recipes/rust_runner_binary.py
@@ -11,7 +11,7 @@ from pathlib import Path
 
 logger = logging.getLogger(__name__)
 
-MIN_RUNNER_VERSION = "0.3.4"
+MIN_RUNNER_VERSION = "0.3.5"
 _REPO_URL = "https://github.com/rysweet/amplihack-recipe-runner"
 
 

